website/docs/podman/accessing-podman-from-another-wsl-instance.md

```diff
@@ -0,0 +1,288 @@
+   ```
+
+   <details>
+   <summary>
+   On a custom Podman Machine, the remote Podman Machine destination might be different.
+
+   Find your Podman Machine name and connection path:
+   </summary>
+   <div>
+
+   1. Identify the sockets available in your WSL distribution.
+
+      The Podman machine shares sockets in a `/mnt/wsl/podman-sockets/` subdirectory named after the Podman machine name.
+
+      In your WSL session, list the available sockets:
+
+      ```shell-session
+      $ find /mnt/wsl/podman-sockets/ -name '*.sock'
+      ```
+
+      Each Podman Machine has a socket for:
+
+      - Rootful Podman: `podman-root.sock`
+      - Rootless Podman: `podman-user.sock`
+
+      Sample output:
+
+      ```shell-session
+      /mnt/wsl/podman-sockets/podman-machine-default/podman-root.sock
+      /mnt/wsl/podman-sockets/podman-machine-default/podman-user.sock
+      ```
+
+   2. Identify the socket that Podman Desktop uses.
+
+      Podman Desktop defaults to rootful Podman.
+      However, consider identifying the active socket.
+
+      The active socket is the default Podman system connection in your Windows session.
+
+      Open a new Command Prompt, and list your Podman system connections:
+
+      ```shell-session
+      > podman system connection list
+      ```
+
+      The default connection line ends with `true`.
+
+      Identify your Podman Machine socket by its URI in Windows:
+
+      - Rootful Podman: `ssh://root@127.0.0.1:59292/run/podman/podman.sock`
+      - Rootless Podman: `ssh://user@127.0.0.1:59292/run/user/1000/podman/podman.sock`
+
+      Sample output:
+
+      ```shell-session
+      Name                         URI                                                          Identity                                                  Default
+
+      podman-machine-default ssh://user@127.0.0.1:59292/run/user/1000/podman/podman.sock C:\Users\Podman Desktop User\.ssh\podman-machine-default false
+      podman-machine-default-root ssh://root@127.0.0.1:59292/run/podman/podman.sock C:\Users\Podman Desktop User\.ssh\podman-machine-default true
+      ```
+
+   3. To define the Podman machine remote destination, prepend with `unix://` the socket path that is available in your WSL, and corresponds to the Podman Desktop active socket:
+
+      For the default Podman machine:
+
+      - Rootful Podman: `unix:///mnt/wsl/podman-sockets/podman-machine-default/podman-root.sock`
+      - Rootless Podman: `unix:///mnt/wsl/podman-sockets/podman-machine-default/podman-user.sock`
+
+   </div>
+   </details>
+
+1. Podman requires you to actively confirm that you want to use a remote connection.
+
+   To always enable the remote connection, set an alias:
+
+   ```shell-session
+   $ alias podman='podman --remote'
+   ```
+
+   :::tip
+
+   On a custom WSL distribution with a Podman client version 4, you might skip this step.
```

Comment:
I am trying to reproduce this to see if there is a more precise version where it starts working as intended. I will first test with Debian bookworm that has Podman 4.3.1.

Reply:
Indeed, on Debian, with Podman 4.3.1:

```shell-session
$ groups
debian adm uucp cdrom sudo dip plugdev users
$ podman system connection list
Name                         URI                                                                     Identity    Default
podman-machine-default-root  unix:///mnt/wsl/podman-sockets/podman-machine-default/podman-root.sock              true
$ podman version
Client:       Podman Engine
Version:      4.3.1
API Version:  4.3.1
Go Version:   go1.19.8
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
```

Reply:
On Podman 4.3.1, when you set the default connection to a remote connection, you need to explicitly set the remote flag (--remote or in the configuration file). On Podman 4.9.0 the remote connection is automatically set (implicitly). When I dig in Podman release notes, I don't see a mention about when this feature has been introduced. I know for sure that it is between 4.3.1 and 4.9.0, but I would like the exact version.

Reply:
I was looking into this recently, and as far as I can tell the only options to trigger remote mode are the config file, the `--remote` flag, the `--url` flag, or the `CONTAINER_HOST|CONTAINER_CONNECTION` environment variables.

https://github.com/containers/podman/blob/caea1f43ed0fdec3fa8785cdb48add99068aceae/cmd/podman/registry/remote.go#L31-L56

(the other flags in the file are hidden ones for docker compatibility afaict)

Reply:
oh, and also the `--connection` option of course

Reply:
On openSUSE Leap, Podman is 4.8.3 and requires the `--remote` flag too.

Reply:
On Oracle Linux, I see we have other issues. Group is already set, but:

```
$ podman system connection list
Name                         URI                                                                     Identity    Default
podman-machine-default-root  unix:///mnt/wsl/podman-sockets/podman-machine-default/podman-root.sock              true
$ podman version
WARN[0000] "/" is not a shared mount, this could cause issues or missing mounts with rootless containers
ERRO[0000] running `/usr/bin/newuidmap 170 0 1000 1 1 100000 65536`: newuidmap: write to uid_map failed: Operation not permitted
Error: cannot set up namespace using "/usr/bin/newuidmap": should have setuid or have filecaps setuid: exit status 1
```